File: Android/java/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/layoutborder/LayoutBorderKit.java
package com.didichuxing.doraemonkit.kit.layoutborder;

import android.app.Activity;
import android.content.Context;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.config.LayoutBorderConfig;
import com.didichuxing.doraemonkit.kit.AbstractKit;
import com.didichuxing.doraemonkit.kit.Category;
import com.didichuxing.doraemonkit.kit.core.DokitIntent;
import com.didichuxing.doraemonkit.kit.core.DokitViewManager;
import com.didichuxing.doraemonkit.kit.core.SimpleDokitStarter;
import com.google.auto.service.AutoService;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by wanglikun on 2019/1/7
 */
@AutoService(AbstractKit.class)
public class LayoutBorderKit extends AbstractKit {


    @Override
    public int getName() {
        return R.string.dk_kit_layout_border;
    }

    @Override
    public int getIcon() {
        return R.mipmap.dk_view_border;
    }

    @Override
    public boolean onClickWithReturn(@NotNull Activity activity) {

        SimpleDokitStarter.startFloating(LayoutLevelDokitView.class);
        LayoutBorderManager.getInstance().start();
        LayoutBorderConfig.setLayoutBorderOpen(true);
        LayoutBorderConfig.setLayoutLevelOpen(true);
        return true;
    }


    @Override
    public void onAppInit(Context context) {
        LayoutBorderConfig.setLayoutBorderOpen(false);
        LayoutBorderConfig.setLayoutLevelOpen(false);
    }

    @Override
    public boolean isInnerKit() {
        return true;
    }


    @Override
    public String innerKitId() {
        return "dokit_sdk_ui_ck_border";
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/layoutborder/LayoutBorderKit.kt
package com.didichuxing.doraemonkit.kit.layoutborder

import android.content.Context
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.AbstractKit
import com.didichuxing.doraemonkit.kit.core.DokitViewManager

/**
 * Created by wanglikun on 2019/1/7
 */
class LayoutBorderKit : AbstractKit() {
    override val name: Int
        get() = R.string.dk_kit_layout_border

    override val icon: Int
        get() = R.mipmap.dk_view_border

    override fun onClick(context: Context?) {
        DokitViewManager.instance.detachToolPanel()
        LayoutManager.instance.showBorder()
    }

    override fun onAppInit(context: Context?) {
    }

    override val isInnerKit: Boolean
        get() = true

    override fun innerKitId(): String {
        return "dokit_sdk_ui_ck_border"
    }
}